tests/test_route.py

```diff
@@ -623,3 +585,94 @@ def test_RouteAndNexthopInDifferentVrf(self, dvs, testlog):
+        self.set_admin_status("Ethernet0", "up")
+        self.set_admin_status("Ethernet4", "up")
+
+        # set ip address and default route
+        dvs.servers[0].runcmd("ip address add 10.0.0.1/31 dev eth0")
+        dvs.servers[0].runcmd("ip route add default via 10.0.0.0")
+
+        dvs.servers[1].runcmd("ip address add 10.0.0.3/31 dev eth0")
+        dvs.servers[1].runcmd("ip route add default via 10.0.0.2")
+
+        fieldValues = [{"nexthop": "10.0.0.1", "ifname": "Ethernet0"}, {"nexthop": "10.0.0.3", "ifname": "Ethernet4"}]
+
+        # get neighbor and arp entry
+        dvs.servers[0].runcmd("ping -c 1 10.0.0.3")
+
+        # get number of routes before adding new routes
+        startNumRoutes = len(self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY"))
+
+        # add route entries
+        timeStart = time.time()
+        for i in range(numRoutes):
+            self.create_route_entry(addrs[i], fieldValues[i % 2])
+
+        # wait until all routes are added into ASIC database
+        self.adb.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY", startNumRoutes + numRoutes) # default timeout is 5 seconds
+        print("Time to add %d routes is %.2f seconds. " % (numRoutes, time.time() - timeStart))
+
+        # confirm all routes are added
+        asicAddrs = set()
+        for key in self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY"):
+            route = json.loads(key)
+            asicAddrs.add(route["dest"])
+        for addr in addrs:
+            assert addr in asicAddrs
+
+        #remove route entries
+        timeStart = time.time()
+        for i in range(numRoutes):
+            self.remove_route_entry(addrs[i])
+
+        # wait until all routes are removed from ASIC database
+        self.adb.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY", startNumRoutes) # default timeout is 5 seconds
+        print("Time to remove %d routes is %.2f seconds. " % (numRoutes, time.time() - timeStart))
+
+        # confirm all routes are removed
+        asicAddrs = set()
+        for key in self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY"):
+            route = json.loads(key)
+            asicAddrs.add(route["dest"])
+        for addr in addrs:
+            assert not addr in asicAddrs
+
+        # remove ip address
+        self.remove_ip_address("Ethernet0", "10.0.0.0/31")
+        self.remove_ip_address("Ethernet4", "10.0.0.2/31")
+
+        # remove l3 interface
+        self.remove_l3_intf("Ethernet0")
+        self.remove_l3_intf("Ethernet4")
+
+        self.set_admin_status("Ethernet0", "down")
+        self.set_admin_status("Ethernet4", "down")
+
+        # remove ip address and default route
+        dvs.servers[0].runcmd("ip route del default dev eth0")
+        dvs.servers[0].runcmd("ip address del 10.0.0.1/31 dev eth0")
+
+        dvs.servers[1].runcmd("ip route del default dev eth0")
+        dvs.servers[1].runcmd("ip address del 10.0.0.3/31 dev eth0")
```

Comment:
>) [](start = 67, length = 1)

Could you provide some sample output in PR description?

Reply:
Added a sample out of the performance test.